common/src/main/java/com/velobank/common/utils/validator/Validator.java

```
@@ -0,0 +1,139 @@
+
+/**
+ * A simple Validator builder class for validating String values.
+ */
+public class Validator {
+
+    private String text;
+    private List<BaseRule> rules;
+    private boolean isValid;
+    private String errorMessage;
+    private ValidationErrorCallback errorCallback;
+    private ValidationSuccessCallback successCallback;
+
+    /**
+     * This constructor will take in a String non-null input value that will be used to validate based on rules applied
+     *
+     * @param text  from which to validate
+     */
+    public Validator(@NonNull String text) {
+        this.text = text;
+        rules = new ArrayList<>();
+        isValid = true;
+    }
+
+    /**
+     * This method validates the text based on the rules applied and sets the error
+     * message if a rule is violated.
+     *
+     * <p>
+     *  If no rules are broken and the {@link ValidationSuccessCallback} is not null, then
+     *  the callback will be invoked.
+     * <p>
+     *  If one rule is broken and the {@link ValidationErrorCallback} is not null, then
+     *  the callback will be invoked.
+     *
+     * @return isValid, which can be true if not rules are broken or false if a rule is broken.
+     */
+    public boolean validate() {
+        for(BaseRule rule : rules) {
+            if(!rule.validate(text)) {
+                setErrorMessage(rule.getErrorMessage());
+                break;
+            }
+
+            if(isValid) {
+                if(successCallback != null)
+                    successCallback.onSuccess();
+            } else {
+                if(errorCallback != null)
+                    errorCallback.onError(errorMessage);
+            }
+        }
+
+        return isValid;
+    }
+
+    /**
+     * This method will add any non-null rule that implements {@link BaseRule} to the list of rules.
+     *
+     * @param rule  from which to add to the list of rules for validation.
+     */
+    public Validator addRule(@NonNull BaseRule rule) {
```

There should be two API expose for initializing the Validator module. First one to set the rules and second to get the instance. 
Also Validator module should throw exception if rules are not set by consumer since it depends upon consumer what rule they want to apply and this module should not provide any default rule.

---

As discussed on my first comment there is no need to have a single instance of this Validator class due to the amount of different rules applied to the UI.

Added a NullPointerException to be thrown when rules are not applied to this Validator.